ui/components/report.jsx

```diff
@@ -1,8 +1,31 @@
-var React = require('react');
+var React = require('react'),
+    Header = require('./header.jsx');
 
 var Report = React.createClass({
   render: function() {
-    return null;
+    var suites = this.props.data,
+        passes = [],
+        failures = [],
+        duration = 0;
+
+    suites.forEach(function(suite) {
+      passes = passes.concat(suite.tests.filter(function(test) {
+        return test.state === 'passed';
+      }));
+
+      failures = failures.concat(suite.tests.filter(function(test) {
+        return test.state !== 'passed';
```

Comment:
Remove duplication by either hoisting the string literal into a constant or by refactoring the check into a private function.